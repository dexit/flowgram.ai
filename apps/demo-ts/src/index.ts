export * from './types'
export * from './runtime'
export * from './translate'
export * from './validate'
export * from './util'
