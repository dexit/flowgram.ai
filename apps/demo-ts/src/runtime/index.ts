export * from './run'
export * from './RuntimeError'
