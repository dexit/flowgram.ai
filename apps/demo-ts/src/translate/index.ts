export * from './nodes'
export * from './translate'
