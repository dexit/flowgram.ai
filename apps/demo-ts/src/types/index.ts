export * from '@flow-ide-editor/schema'
export * from './nodes'

export class InvalidSchemaError extends Error {}
export class InvalidNodeTypeError extends Error {}
export class InvalidTypeError extends Error {}
export class NotImplementedError extends Error {}

export type FlowArgs = Record<string, unknown>
