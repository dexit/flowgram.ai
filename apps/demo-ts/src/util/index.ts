export * from './log'
export * from './JSBuffer'
