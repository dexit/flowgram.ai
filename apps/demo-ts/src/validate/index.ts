import { FlowNodeSchema } from '@flow-ide-editor/schema'
import { InvalidSchemaError } from '../types'

// TODO use json-schema validation
export function validateOrThrow(node: FlowNodeSchema): void {
  if (!node.definition) throw new InvalidSchemaError()
  if (!node.definition.info) throw new InvalidSchemaError()
  if (!node.definition.info.title) throw new InvalidSchemaError()
}
