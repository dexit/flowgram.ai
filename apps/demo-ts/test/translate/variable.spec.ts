import { VariableNodeSchema, translateNode } from '../../src'

describe('translate', () => {
  it('Variable - 0 param', () => {
    const node: VariableNodeSchema = {
      id: 'variable_dd605e4e286',
      type: 'variable',
      definition: {
        info: { title: 'variable', description: '', icon: '' },
        outputs: {
          type: 'object',
          properties: {},
        },
        script: { language: 'python', content: '' },
      },
      data: {
        setVariables: {},
      },
    }

    expect(translateNode(node)).toMatchSnapshot()
  })

  it('Variable - 1 param', () => {
    const node: VariableNodeSchema = {
      id: 'variable_dd605e4e286',
      type: 'variable',
      definition: {
        info: { title: 'variable', description: '', icon: '' },
        outputs: {
          type: 'object',
          properties: {
            a: { type: 'string' },
          },
        },
        script: { language: 'python', content: '' },
      },
      data: {
        setVariables: {
          '0': {
            left: { type: 'ref', content: '@.a' },
            right: { type: 'expression', content: 'start.outputs.query' },
            operator: 'assign',
          },
        },
      },
    }

    expect(translateNode(node)).toMatchSnapshot()
  })

  it('Variable - more params', () => {
    const node: VariableNodeSchema = {
      id: 'variable_dd605e4e286',
      type: 'variable',
      definition: {
        info: { title: 'variable', description: '', icon: '' },
        outputs: {
          type: 'object',
          properties: {
            v_string: { type: 'string' },
            v_integer: { type: 'integer' },
            v_number: { type: 'number' },
            v_boolean: { type: 'boolean' },
            v_array: { type: 'array', items: { type: 'string' } },
            v_object: { type: 'object', properties: { a: { type: 'number' } } },
            v_map: { type: 'map', additionalProperties: { type: 'string' } },
          },
        },
        script: { language: 'python', content: '' },
      },
      data: {
        setVariables: {
          '0': {
            left: { type: 'ref', content: '@.v_string' },
            right: { type: 'expression', content: "'abc'" },
            operator: 'assign',
          },
          '1': {
            left: { type: 'ref', content: '@.v_integer' },
            right: { type: 'expression', content: '10' },
            operator: 'assign',
          },
          '2': {
            left: { type: 'ref', content: '@.v_number' },
            right: { type: 'expression', content: '3.14' },
            operator: 'assign',
          },
          '3': {
            left: { type: 'ref', content: '@.v_boolean' },
            right: { type: 'expression', content: 'True' },
            operator: 'assign',
          },
          '4': {
            left: { type: 'ref', content: '@.v_array' },
            right: { type: 'expression', content: "['a','b']" },
            operator: 'assign',
          },
          '5': {
            left: { type: 'ref', content: '@.v_object' },
            right: { type: 'expression', content: "{'a':1}" },
            operator: 'assign',
          },
          '6': {
            left: { type: 'ref', content: '@.v_map' },
            right: { type: 'expression', content: "{'b':2}" },
            operator: 'assign',
          },
        },
      },
    }

    expect(translateNode(node)).toMatchSnapshot()
  })
})
