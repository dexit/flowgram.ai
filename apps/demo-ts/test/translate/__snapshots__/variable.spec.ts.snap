// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`translate Variable - 0 param 1`] = `
"  //=== variable@variable | {level:1,blockIndex:0}
  const variable = {}
  variable.outputs = {}
  "
`;

exports[`translate Variable - 1 param 1`] = `
"  //=== variable@variable | {level:1,blockIndex:0}
  const variable = {}
  variable.outputs = {
    a: '',
  }
  variable.outputs.a = start.outputs.query
  "
`;

exports[`translate Variable - more params 1`] = `
"  //=== variable@variable | {level:1,blockIndex:0}
  const variable = {}
  variable.outputs = {
    v_string: '',
    v_integer: 0,
    v_number: 0,
    v_boolean: false,
    v_array: [],
    v_object: {
      a: 0,
    },
    v_map: {},
  }
  variable.outputs.v_string = 'abc'
  variable.outputs.v_integer = 10
  variable.outputs.v_number = 3.14
  variable.outputs.v_boolean = True
  variable.outputs.v_array = ['a','b']
  variable.outputs.v_object = {'a':1}
  variable.outputs.v_map = {'b':2}
  "
`;
